### PR TITLE
Apply review findings from PR #3844's end-of-work round

### PR DESCRIPTION
**Pull request by tinnhlt-gra**
## Summary

PR #3844 (QC Wave 3) auto-merged before the required `craft-reviewer` and `security-redteam` end-of-work agents finished reviewing it (both were still running when GitHub's own required checks all went green). This PR applies their confirmed findings, plus one duplicate-fix cleanup discovered while investigating the merged state.

**craft-reviewer findings applied:**
- `deals.Store.Figures` no longer re-spells `CloseIsOverdue`'s own comparison in SQL. `closedate.go` says plainly it is the ONE place that comparison is made; the SQL boolean was a second, silent copy of it that could drift from the original. `Figures` now fetches candidate rows and calls `CloseIsOverdue` in Go — the same shape `closedatesweep.go` already uses.
- `classifyRisk`'s `ExpectedMinorBase` guard now also requires a named base currency (`money.base != ""`), matching the sibling guard `dayMoney.value` already holds for the identical inconsistent-FX-seam edge case.
- Dangling test cross-reference fixed (`promotedPersonID` → `decodeLeadPromoted`, renamed in the original PR).
- The clock inconsistency between the two "close date overdue" integration tests is resolved — both now seed from Go's own clock, matching what the redesigned `Figures` implementation actually compares against.
- Trigger literals named consistently (`activityCapturedTrigger`, `leadPromotedTrigger`, `leadDisqualifiedTrigger`).
- The shared completion helper's behavioral doc moved off `CompleteOpenSystemTasksForLead` onto `completeOpenSystemTasksLinkedBy`, so the new `CompleteOpenSystemTasksForPerson` doesn't read with none of it.
- `WorklistDealFacts.expected_minor_base`'s contract description no longer promises a `win_probability` weighting nothing in this pipeline computes (confirmed: `win_probability` is itself never populated anywhere in `compose/attention`).

**security-redteam findings applied:**
- `RequirePublicOrigin`'s development/test exemption is now exact-empty (`raw == ""`) rather than trimmed-empty. A whitespace-only value reads as "configured" by the sending caller's own check (`footerlanguage.go`, untrimmed) — admitting it here as "unset" would have let both gates agree to send a tokenized link built from whitespace.
- `CompleteOpenSystemTasksForPerson` (margince#3764's fix) now bounds its completion to tasks that existed by the triggering event's own `OccurredAt`. The handler runs asynchronously off the outbox, so "a fresh person cannot yet carry anything else" was only true at the instant of promotion — a sibling automation (`no_activity_reminder`, `check_in_cadence`) could anchor its own system task on the same freshly-created person before this handler actually runs, and the unbounded query would have completed that task too, with an audit row claiming a follow-up that never happened.

**Duplicate-fix cleanup (discovered independently, not one of the two reviews' findings):**
- PR #3834 landed a fix for margince#3800 (a briefing entry whose deal cannot be resolved) at a better layer — `attentionBriefing.Queue`, covering both `/attention` and `/worklist` — while PR #3844 was still in flight on the same commit range. Both merged. `nameTheMoney`'s narrower, worklist-only drop logic from #3844 is now removed; #3834's fix is the sole owner of this invariant, which is exactly the two-answers-to-one-question drift AGENTS.md's reuse principles warn about.

## Deliberately not fixed

- `dedupe_outcome` (the `lead.promoted` event field the merged-vs-created gate reads) stays a bare string rather than becoming a typed OpenAPI enum. `public-events.yaml` has a DOCUMENTED house rule against inline enums for exactly this reason: the generator emits package-scope Go constants named after the enum values, which can collide with another schema's same-named values. "created"/"merged" are generic enough that verifying no collision exists needs more care than this follow-up's scope; tracked as the lowest-severity finding, not blocking.
- The merged-promotion sub-case of margince#3764 (still open, filed as margince#3843) is unchanged — the craft-reviewer's suggestion to "open a labelled issue and cite it here" was not applied verbatim: this codebase has zero existing `// TODO` comments citing issues, and the code already states the gap as a plain, test-backed fact rather than rationalizing it, which is what AGENTS.md's own rule asks for.

## Test plan

- [x] `make check` — full local gate, green.
- [x] `craft static --strict` (diff-scoped) — 0 blocker/major/minor.
- [x] `make test-it DIR=backend/internal/modules/activities` and `DIR=backend/internal/compose/integration RUN=TestDealFigures` — real-Postgres integration lanes, green, including the new `TestAPromotedLeadDoesNotCompleteATaskMintedAfterThePromotionEvent` regression guard.
- [x] Confirmed the `internal/compose` package's unrelated overlay/HubSpot test failures when run standalone are a pre-existing test-harness env gap (`MARGINCE_TEST_REDIS not set`), not a regression from this diff.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01AzkBShFNemkBKN3h5faNxf

---
## Summary by cubic
Applies the confirmed review fixes from QC Wave 3 and removes overlapping deal-resolution logic to prevent incorrect task completion, overdue status, and currency-risk values.

**Bug Fixes**
- Person-linked follow-up completion now excludes system tasks created after the promotion event.
- Whitespace-only public origins are rejected in development and test environments.
- FX conversions without a named base currency no longer set `expected_minor_base` or a materiality verdict.

**Refactors**
- `deals.Store.Figures` now uses the shared `CloseIsOverdue` comparison with the installation timezone instead of duplicating it in SQL.
- Unresolved deal rows remain unchanged in `nameTheMoney`; the briefing queue owns filtering them.
- Regenerated API schemas and stabilized close-date integration tests against clock and session-timezone boundaries.

<sup>Written for commit c39d98bb2f13bdc0ea8ae2baa54a85d9dc5f81ee. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/margince/margince/pull/3921?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

- **New Features**
  - Added analytics schema, query, and explanation endpoints for human-only analytics.
  - Added communication evidence and contextual metadata to email workflows.
  - Added relationship facts and assignee ownership details to attention items.

- **Bug Fixes**
  - Deal figures now avoid displaying converted values without a known base currency.
  - Unresolved deals remain visible without incomplete figures.
  - Follow-up tasks respect promotion-event timing.
  - Close-date overdue status respects the installation timezone.
  - Whitespace-only origins are validated instead of treated as unset.

- **Documentation**
  - Clarified expected minor base values are not weighted by win probability.
  - Clarified worklist totals and category counts cover the full candidate population.

### CODE REVIEW

**Comment by tinnhlt-gra** [comment]
Addressed both CodeRabbit findings in 326d06aae:
- `classifyrisk.go` line 44: an unnamed conversion (base="", byItem populated) no longer drives `known`/the material verdict — same guard `ExpectedMinorBase` already had.
- `dealfigures_integration_test.go` line 199: pinned `e.Deals.WithClock` to the same instant used to seed `expected_close_date`, closing the UTC-midnight race.

**Comment by tinnhlt-gra** [comment]
Applied the remaining confirmed findings from the end-of-work craft-reviewer + security-redteam round (both converged independently on the hand-typed placeholder) in 455126292:
- `followupresolve.go`: derived `$6` in `completeOpenSystemTasksLinkedBy`'s conditional bound from the args slice via the same `arg()` closure `dealfigures.go` uses, instead of a hand-typed literal.
- `classifyrisk.go`: dropped the now-redundant `money.base != ""` clause on `ExpectedMinorBase` — `expectedRevenue` (previous commit) already answers `known=false` for that case, so it could never be the deciding term.
- Extracted `installationZone` (now in `dealfigures.go`) so `dealfigures.go` and `closedatesweep.go` share one `time.LoadLocation`-plus-error-wrap spelling instead of two, and fixed a comment claiming `installationToday` reads the zone "the same way" — it never loads a `*time.Location` at all.
- `dealfigures_integration_test.go`: seeded close date bound as an explicit calendar-date string, closing a latent (never actually observed) dependency on the test connection's session TimeZone.

Not applying: the two-clock-domain concern both reviewers raised (`ev.OccurredAt`, app clock, compared against `activity.created_at`, DB clock) — a properly-derived fix needs a DB-clock-consistent anchor threaded through the workflow engine's event dispatch, which reaches beyond this module. Recording as a follow-up rather than guessing an underived slack constant under this round's budget.

File: backend/internal/modules/activities/followupresolve.go [hunk 1]
```diff
@@ -271,40 +305,40 @@ func (w followUpAutoResolve) linkedLeads(ctx context.Context, activityID ids.Act
...
 			SELECT a.id, a.version FROM activity a
 			JOIN activity_link l ON l.activity_id = a.id
 			WHERE l.%s = $1 AND a.kind = $2 AND a.source = $3
 			  AND (a.captured_by = $4 OR a.captured_by LIKE $5)
-			  AND a.is_done = false AND a.archived_at IS NULL
-			ORDER BY a.id`, column), linkValue, string(crmcontracts.ActivityKindTask), systemSource,
-			systemCapturedBy, systemCapturedByPattern)
+			  AND a.is_done = false AND a.archived_at IS NULL`, column)
+		if before != nil {
+			query += storekit.SQLf(" AND a.created_at <= $%d", arg(*before))
```
[the remark this answers is not in the document]
Comment (tinnhlt-gra) [reply backend/internal/modules/activities/followupresolve.go:338]: Confirmed — same clock-domain gap craft-reviewer and security-redteam both flagged independently in the earlier review round on this PR (see the PR body/comments above). Not fixing it here: a properly-derived fix needs a DB-clock-consistent anchor for the promotion event (e.g. reading back `event_outbox.created_at` for this event, or a lead.promoted_at column), which touches `workflow.Event`'s construction and every handler that reads `OccurredAt` — architecture-level, beyond this round's scope per AGENTS.md's carve-out for fixes needing a broader design decision. Tracked as margince#3933 with the concrete fix options.
